### ISSUE TITLE
[Bug]: Missing rule for removing leading/trailing spaces in className

### ISSUE THREAD

**Issue by Kyzegs** [jsx-eslint/eslint-plugin-react#3574]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

There is no rule for removing leading/trailing white space in className.

```jsx
// Incorrect
<div className=" w-100 d-flex flex-row   justify-between   align-content-center   "></div>
```

```jsx
// Correct
<div className="w-100 d-flex flex-row justify-between align-content-center"></div>
```

### Expected Behavior

For there to be a rule for this.

### eslint-plugin-react version

v7.32.2

### eslint version

v17.0.2

### node version

v16.17.0

**Comment by ljharb** [comment]
I'm not sure why we'd want to special-case ``className` like that - in my experience it's usually set to the return of https://www.npmjs.com/package/classnames rather than manually typing strings anyways.

**Comment by Kyzegs** [comment]
How do you mean special-case? It's just so that there are no excess spaces?

**Comment by ljharb** [comment]
I mean that this wouldn’t be a generically applicable rule to any props with string values - so className would be a special case in that it would be hardcoded in the rule.

**Comment by Kyzegs** [comment]
Ah, yeah! But I would say it makes most sense for `className`.

**Comment by ljharb** [comment]
I agree - except that my experience tells me it's a best practice to use a utility and not to manually type out a space-separated string, and such a utility would have trimming built in already.
